src/pytorch_lightning/loggers/base.py

```diff
@@ -12,16 +12,20 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from typing import Any, Callable, Mapping, Optional, Sequence
+
+import numpy as np
+
 import pytorch_lightning.loggers.logger as logger
 from pytorch_lightning.utilities.warnings import rank_zero_deprecation
 
 
-def rank_zero_experiment(*args, **kwargs) -> None:  # type: ignore[no-untyped-def]
+def rank_zero_experiment(fn: Callable) -> Any:
```

Comment:
```suggestion
def rank_zero_experiment(fn: Callable) -> Callable:
```
this return the wrapped function?

Reply:
IDK if it returns the wrapped function; but it does return the inner `experiment` function of loggers.logger.rank_zero_experiment ... so your suggestion is correct.